bigtable/client/internal/table_admin.cc

```diff
@@ -190,6 +190,31 @@ bool TableAdmin::CheckConsistency(
   return response.consistent();
 }
 
+bool TableAdmin::WaitForConsistencyCheckHelper(
+    bigtable::TableId const& table_id,
+    bigtable::ConsistencyToken const& consistency_token) {
+  btproto::CheckConsistencyRequest request;
+  request.set_name(TableName(table_id.get()));
+  request.set_consistency_token(consistency_token.get());
+  MetadataUpdatePolicy metadata_update_policy(
+      instance_name(), MetadataParamTypes::NAME, table_id.get());
+
+  grpc::Status status;
+  do {
+    auto response = ClientUtils::MakeCall(
+        *client_, rpc_retry_policy_->clone(), rpc_backoff_policy_->clone(),
+        metadata_update_policy, &AdminClient::CheckConsistency, request,
+        "CheckConsistency", status, true);
+
+    if (response.consistent()) {
+      return true;
+    }
+
+  } while (not polling_policy_->Exhausted());
+
+  return false;
```

Comment:
Consider this instead:

```C++
bool TableAdmin::WaitForConsistencyCheckHelper(
    bigtable::TableId const& table_id,
    bigtable::ConsistencyToken const& consistency_token,
    grpc::Status& status) {
auto polling_policy = polling_policy_->clone();
do {
} while (not polling_policy->Exhausted());
  return false;
}
```

This way the `status` parameter carries any errors for the last operation, and you can raise in the caller if there is an error, but return false if there is none.